File: backend/app/schema/product.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.schema.product_image import ProductImageRead


class ProductBase(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    category: str = Field(min_length=1, max_length=100)
    price: Decimal = Field(gt=0, decimal_places=2)
    description: str | None = None


class ProductCreate(ProductBase):
    is_active: bool = True


class ProductUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    category: str | None = Field(default=None, min_length=1, max_length=100)
    price: Decimal | None = Field(default=None, gt=0, decimal_places=2)
    description: str | None = None
    is_active: bool | None = None


class ProductRead(ProductBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime
    images: list[ProductImageRead] = []

File: admin-panel/src/types/product.ts
export type Product = {
    id: string
    title: string
    category: string
    price: string
    image_url: string | null
    description: string | null
    is_active: boolean
    created_at: string
    updated_at: string
}

export type ProductInput = {
    title: string
    category: string
    price: string
    image_url?: string | null
    description?: string | null
    is_active?: boolean
}
